mods/prevent-focus-stealing.wh.cpp

```diff
@@ -0,0 +1,371 @@
+- `Never Focus` prevents manual activation, so it should only be enabled if you never want to interact with the window.
+
+​
+
+---
+
+​
+
+## Preventing focus being stolen by Chromium (Electron / Browsers)
+
+The Chromium DevTools window is especially annoying, as it opens asynchronously and has to be reopened every time you apply changes while working on development of a Chrome extension or an Electron-based app. You can now auto reload changes and have one or more DevTools windows reopen without your editor losing focus.
+
+### Application Executables
+
+Make sure `Application Executables` includes the process you are using (such as `electron.exe`, `brave.exe` or `chrome.exe`).
+
+### Required Arguments
+
+You can leave `Required Arguments` blank unless you only want to target a specific instance of an app, such as an electron app which you launch with custom argument during development.
+
+### Excluded Arguments
+
+You should set `Excluded Arguments` to `--type` for efficiency so that only the main chromium process which creates windows is hooked.
+
+### Window Title
+
+- Leave title blank to prevent initial focus of all windows for the application.
+- `Developer Tools` will target all DevTools windows.
+- `Developer Tools - https://example.com` will target only DevTools for that site.
+- `Developer Tools - chrome-extension://<extension_id>` will target only DevTools windows for that extension.
+
+### Never Focus
+
+Leave `Never Focus` disabled so that you can manually activate the window and interact with it.
+
+*/
+// ==/WindhawkModReadme==
+
+// ==WindhawkModSettings==
+/*
+- applications:
+  - - exes: [electron.exe, brave.exe, chrome.exe]
+      $name: Application Executables
+      $description: "Executables cannot be blank. Each executable can be either a file name (e.g. electron.exe), full path (e.g. C:\\Program Files\\Brave\\brave.exe), or path glob (e.g. *electron.exe)."
+    - args: ['']
+      $name: Required Arguments
+      $description: "Each argument is a substring to match in the command line. All must match. Leave empty to include all processes."
+    - exclude_args: [--type]
+      $name: Excluded Arguments
+      $description: "Each argument is a substring to exclude from the command line. Any match excludes the process. Leave empty to include all processes."
+    - title: Developer Tools
+      $name: Window Title
+      $description: "The window title prefix to match. Leave blank to match all windows for the process."
+    - never_focus: false
+      $name: Never Focus
+      $description: "Never allow the window to be become focused, instead of only when it initially opens."
+  $name: Application Windows
+  $description: "Prevent focus being stolen by these application windows."
+*/
+// ==/WindhawkModSettings==
+
+#include <windhawk_utils.h>
+#include <windows.h>
+#include <shlwapi.h>
+#include <vector>
+#include <set>
+#include <map>
+
+#define WM_USER_REMOVE_NOACTIVATE (WM_USER + 1)
```

Comment:
It's not safe to assume that this message isn't used by windows for their own needs. Consider using `RegisterWindowMessage` instead.

But do you even need it? Why not just call `SetWindowLongPtrW(hWnd, GWL_EXSTYLE, ...)` from another thread?

Reply:
This is a good point, will have to address this. Delaying until the next message loop tick was necessary to make it work reliably.